mobile/App.js

```diff
@@ -97,11 +99,17 @@ class App extends React.Component<{}, State> {
                 {!token ? (
                   <Login />
                 ) : (
-                  <CurrentUser>
-                    {({ currentUser }) =>
-                      currentUser && currentUser.id ? <TabBar /> : <Login />
-                    }
-                  </CurrentUser>
+                  <Query query={getCurrentUserCommunityConnectionQuery}>
+                    {({ data: { networkStatus, user }, refetch }) => {
+                      if (networkStatus === 1 || networkStatus === 2)
+                        return null;
+                      if (!user) return <Login />;
+                      if (!user.username) return <SetUsername />;
+                      if (user.communityConnection.edges.length === 0)
+                        return <ExploreCommunities refetch={refetch} />;
```

Comment:
nit: if the `addCommunityMembers` mutation automatically updates the cache locally this refetch thing should be unnecessary

Reply:
Yeah, I'll go into the `options` config for these mutations and make sure they update the cache - as it works today we do need the refetch.

Reply:
Actually...can we postpone this. I'm not sure what unintended side effects this will have on the web side of things. Because it's possible that a user on web lands directly on a community page, joins the community, and there's no `getCurrentUserCommunityConnection` query to update in the cache. On mobile we don't have this problem since the `App.js` ensures that this query will be run *every* time and will exist in the cache.

Let me know what you think. If you agree, this PR is good to go.